app/main.py

```diff
@@ -1,6 +1,17 @@
-from typing import Callable
+from typing import Callable, Any
 
 
 def cache(func: Callable) -> Callable:
-    # Write your code here
-    pass
+    storage = {}
+
+    def wrapper(*args: list) -> Any:
+        if args in storage:
```

Comment:
Using `args` directly as a dictionary key can lead to issues because `args` is a tuple, which is hashable, but if any of its elements are mutable (like lists), it will cause a `TypeError`. Consider converting `args` to a hashable type that only contains immutable elements, or ensure that the function only accepts immutable arguments.